### PR TITLE
webpack -> rspack migration

### PR DESCRIPTION
**Pull request by Theodlz**
Bundling times for SkyPortal (in production mode for example) are cut in half using rspack, and it looks like the bundle sizes are also slightly reduced in production mode.

Overall rspack seems to offer us the same features as webpack, while speeding up things which is very handy while developing, and while running any CI/CD.

I'd be in favor of moving to rspack, especially given how minimal the changes needed are.

### CODE REVIEW

**Comment by stefanv** [comment]
If it's a 1-to-1 replacement, that's fine; we have a somewhat complex webpack configuration, so I wanted to make sure that it takes all behavior into consideration.

**Comment by Theodlz** [comment]
@stefanv I had to edit the CI to point to a custom branch of the template app (that will be its own PR), but CI is running and everything is working as expected.

I checked the config, and `rspack` runs some pretty strict validation, which should make it error out if the config contains things that are not supported, instead of ignoring it. So, this should mean that our config is valid. I looked through the rest of the doc, and none of our current config isn't supported. You can see that here: https://rspack.dev/guide/migration/webpack#updating-configuration

**Comment by Theodlz** [comment]
let me know when/if you are happy with this, and I'll remove the temporary changes, and we can look at the PR on the template app.

**Comment by Theodlz** [comment]
https://github.com/cesium-ml/baselayer_template_app/pull/304 Template App PR is ready

https://github.com/skyportal/skyportal/pull/5278 SkyPortal PR is ready, can be merged once baselayer's is.

**Comment by stefanv** [comment]
I'm happy to give this a try. Correct order seems to be: baselayer template app, then updated version of this, then skyportal.

I think it would be good to have a note somewhere that we use rspack as a 1:1 replacement of webpack, in case other projects prefer to use that.

**Comment by Theodlz** [comment]
> I'm happy to give this a try. Correct order seems to be: baselayer template app, then updated version of this, then skyportal.
> 
> I think it would be good to have a note somewhere that we use rspack as a 1:1 replacement of webpack, in case other projects prefer to use that.

That's a great idea. Would you rather have me add this to baselayer, or the template app that has the rspack.config.js?

**Comment by stefanv** [comment]
Baselayer docs, maybe, since that's where the command is being invoked from?

**Comment by Theodlz** [comment]
@stefanv would you rather have me merge baselayer pointing to the edited template app (for the CI to be green on main), then merge the PR on the template app, then open another one on baselayer to go back to the main of the template app?

Or, I get this PR to go back to the main branch of the template app, we merge it and CI fails, but once the template app's PR is merged we just rerun the CI here to have it be green.

Any preference? I feel like the second option is best, since I just need to rerun an action rather than re-open a PR.

**Comment by Theodlz** [comment]
adding the docs changes in a minute btw

**Comment by Theodlz** [comment]
done @stefanv, I added a short paragraph/section dedicated to the web app bundling microservice

**Comment by Theodlz** [comment]
> Baselayer docs, maybe, since that's where the command is being invoked from?

done, I added a short paragraph/section dedicated to the web app bundling microservice

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -43,9 +43,9 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
-      - name: Install template app
+      - name: Install template app (modified, temporary)
         run: |
-          git clone https://github.com/cesium-ml/baselayer_template_app
+          git clone -b rspack-v1 https://github.com/Theodlz/baselayer_template_app
```
Comment (stefanv) [inline .github/workflows/test.yaml:48]: Can merge as soon as this has returned to cesium-ml

**Comment by stefanv** [comment]
Thanks!

**Pull request merged**